File: backend/src/main/kotlin/com/casadosreclamos/dto/NewRequestDto.kt
package com.casadosreclamos.dto

import org.eclipse.microprofile.openapi.annotations.media.Schema

data class NewRequestDto(
    var clientId: Long?,
    var amount: Int?,
    var observations: String?,
    var application: Boolean?,
    var brand: BrandDto?,

    @Schema(oneOf = [OneFace::class, TwoFaces::class, LeftShowcase::class, RightShowcase::class])
    var type: RequestTypeDto?
)

File: frontend/src/models/NewRequestDto.js
export default class NewRequestDto {
  clientId = null // Long
  brandId = null // Long
  materialId = null // Long
  amount = null // Int
  measurements = null // Measurements
  application = null // Boolean
  images = null // RequestTypeDto

  constructor(obj) {
    if (obj) {
      this.clientId = obj.clientId
      this.brandId = obj.brandId
      this.materialId = obj.materialId
      this.amount = obj.amount
      this.measurements = new Measurements(obj.measurements)
      this.application = obj.application
      this.images = RequestTypeDto.from(obj.images)
    }
  }
}